### PR TITLE
Extract interfaces from CrdtSet.Data and CrdtSingleton.Data

### PR DESCRIPTION
**Pull request by jasonwyatt**
This will help keep things tidy when implementing CrdtEntity, since it's a bunch of sets (collections) and singletons.

### CODE REVIEW

File: src_kt/java/arcs/crdt/CrdtSet.kt [hunk 1]
```diff
@@ -18,10 +18,13 @@ import arcs.crdt.internal.VersionMap
 
 /** A [CrdtModel] capable of managing a set of items [T]. */
 class CrdtSet<T : Referencable>(
-  private var _data: Data<T> = Data()
+  /** Initial data. */
+  private var _data: Data<T> = DataImpl(),
```
Comment (cromwellian) [inline src_kt/java/arcs/crdt/CrdtSet.kt]: Is prefixing private variables with _ a Kotlin style thing, or is this a holdover from the Typescript?

File: src_kt/java/arcs/crdt/CrdtSingleton.kt [hunk 2]
```diff
@@ -15,27 +15,45 @@ import arcs.crdt.CrdtSet.Operation.Add
...
 
-  override fun updateData(newData: CrdtSet.Data<T>) = set.updateData(newData)
+  override fun updateData(newData: Data<T>) = set.updateData(newData)
+
+  /** Abstract representation of the data stored by a [CrdtSingleton]. */
+  interface Data<T : Referencable> : CrdtSet.Data<T> {
+    override fun copy(): Data<T>
+  }
+
+  /** Concrete representatino of the data stored by a [CrdtSingleton]. */
```
Comment (cromwellian) [inline src_kt/java/arcs/crdt/CrdtSingleton.kt]: sp representation

File: src_kt/java/arcs/crdt/CrdtSet.kt (hunk 1 above)
Comment (jasonwyatt) [reply src_kt/java/arcs/crdt/CrdtSet.kt]: it's a thing when they're backing-values for a property with the non-underscored version of the name.

File: src_kt/java/arcs/crdt/CrdtSingleton.kt (hunk 2 above)
Comment (jasonwyatt) [reply src_kt/java/arcs/crdt/CrdtSingleton.kt]: Thanks, I guess I felt like writing some fake italian-english hybrid there 🙃

File: src_kt/java/arcs/crdt/CrdtSet.kt (hunk 1 above)
Comment (jasonwyatt) [reply src_kt/java/arcs/crdt/CrdtSet.kt]: [Context:](https://kotlinlang.org/docs/reference/coding-conventions.html#naming-rules) "Names for backing properties."

**Pull request merged**